res/controllers/common-hid-packet-parser.js

```diff
@@ -9,6 +9,24 @@
 this.HIDDebug = function(message) {
     console.log(`HID ${message}`);
 };
+/**
+ * creates a `DataView` from any ArrayBuffer, TypedArray
+ * or plain Array (clamped to 8-bit width).
+ *
+ * @param {number[] | ArrayBuffer | TypedArray} bufferLike Object that can be represented as a sequence of bytes
+ * @returns {DataView} dataview over the bufferLike object
+ */
+const createDataView = function(bufferLike) {
+    return new DataView((() => {
+        if (Array.isArray(bufferLike)) {
+            return new Uint8ClampedArray(bufferLike).buffer;
+        } else if (ArrayBuffer.isView(bufferLike)) {
+            return bufferLike.buffer;
+        } else {
+            return bufferLike;
+        }
+    })());
+};
```

Comment:
I noticed a performance impact by your implementation. I played a bit with the code and this is faster. I guess because the bufferLike is in nearly all cases a Uint8Array.
I don't think there is a need to support any other TypedArray type.

Reply:
>  I don't think there is a need to support any other TypedArray type.

I disagree, and I don't think it would be difficult to support it. 
The improvement from performance is likely just the change in ordering of the if cases, can you experiement whether there is a significant difference when you change `bufferLike.constructor === Uint8Array` back to `ArrayBuffer.isView(bufferLike)` in your suggestion? Also I doubt the IIFE is make any difference either because it should just be inlined by the JIT.

Reply:
Thee IIFE is removed for better code readabilty and I also doubt it affects the performance. I will benchmark your suggestions.

Reply:
> Thee IIFE is removed for better code readabilty

I guess this is certainly something to argue about, but I think the IIFE improves readability because it DRY's up the code. If I see the same code repeated in all branches, it usually creates suspicion for me that this could be the source of a bug from mindlessly copy-pasting code. But that is certainly controversial.